common/common.go

```diff
@@ -0,0 +1,31 @@
+package common
```

Comment:
Is there a reason why we introduce a separate package for this? Shouldn't it be in the controller package or did you have a reason to split it off?